### PR TITLE
Add tests for column selection and renaming for GroupedDataFrame

### PR DESCRIPTION
**Pull request by bkamins**
As @nalimilan spotted there is a risk that these cases were not covered comprehensively.

### CODE REVIEW

File: test/grouping.jl [hunk 1]
```diff
@@ -3230,6 +3230,28 @@ end
...
+    @test combine(gdf, [:x, :y]) ==
+          DataFrame(id=[1, 1, 1, 2, 3, 3], x=[1, 2, 6, 3, 4, 5], y=[11, 12, 16, 13, 14, 15])
+    @test combine(gdf, [:x, :y], :z) ==
+          DataFrame(id=[1, 1, 1, 2, 3, 3], x=[1, 2, 6, 3, 4, 5],
+                    y=[11, 12, 16, 13, 14, 15], z=[21, 22, 26, 23, 24, 25])
+    @test_throws ArgumentError combine(gdf, :x, :x)
+    @test_throws ArgumentError combine(gdf, :x => :y, :y)
+
+    @test select(gdf, :x) == select(df, :id, :x)
+    @test select(gdf, :x => :y ) == select(df, :id, :x => :y)
```
Comment (nalimilan) [inline test/grouping.jl:3248]:
```suggestion
    @test select(gdf, :x => :y) == select(df, :id, :x => :y)
```

**Review by nalimilan — APPROVED** [review]
Thanks!

**Comment by bkamins** [comment]
Thank you! I will merge this wen CI passes and then I think it would be best to rebase the threading PR (as you have to rebase it anyway to switch to `main`). OK?

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**